### PR TITLE
Feature/connect follow unfollow

### PR DESCRIPTION
**Pull request by ArtemLigerko**

### CODE REVIEW

File: src/redux/reducers/user.ts [hunk 1]
```diff
@@ -254,6 +276,30 @@ const userSlice = createSlice({
...
+    builder.addCase(addFollower.pending, (store) => {
+      store.followedUser.isLoading = true;
+    });
+    builder.addCase(addFollower.fulfilled, (store, { payload }) => {
+      store.followedUser = { ...payload };
+      store.followedUser.isLoading = false;
+    });
+    builder.addCase(addFollower.rejected, (store) => {
+      store.followedUser.isLoading = false;
+      store.followedUser.error = "Failed to fetch followimg user";
```
Comment (DavidKorystov) [inline src/redux/reducers/user.ts]:
```suggestion
      store.followedUser.error = "Failed to fetch following user";
```

File: src/redux/reducers/user.ts [hunk 2]
```diff
@@ -254,6 +276,30 @@ const userSlice = createSlice({
...
+    builder.addCase(deleteFollower.pending, (store) => {
+      store.followedUser.isLoading = true;
+    });
+    builder.addCase(deleteFollower.fulfilled, (store, { payload }) => {
+      store.followedUser = { ...payload };
+      store.followedUser.isLoading = false;
+    });
+    builder.addCase(deleteFollower.rejected, (store) => {
+      store.followedUser.isLoading = false;
+      store.followedUser.error = "Failed to fetch followimg user";
```
Comment (DavidKorystov) [inline src/redux/reducers/user.ts]:
```suggestion
      store.followedUser.error = "Failed to fetch following user";
```

**Review by DavidKorystov — APPROVED** [review]
lgtm

File: src/containers/Banner.tsx [hunk 3]
```diff
@@ -151,6 +155,48 @@ const Banner = ({ isEditBanner, user }: BannerProps) => {
...
+      return t("common.follow");
+    }
+  };
+
+  const [buttonText, setButtonText] = useState(initialText);
+
+  const userByUsernameId = useSelector<
+    RootState,
+    RootState["user"]["userByUsername"]["_id"]
+  >((store) => store.user.userByUsername._id);
```
Comment (alexandrtovmach) [inline src/containers/Banner.tsx]:
```
  const currentUser = useSelector<
    RootState,
    RootState["user"]["userByUsername"]
  >((store) => store.user.userByUsername);

// ...

const handleFollowClick = () => {
    if (isFollowed) {
      dispatch(userActions.deleteFollower(currentUser._id));
```

File: src/containers/Banner.tsx [hunk 4]
```diff
@@ -151,6 +155,48 @@ const Banner = ({ isEditBanner, user }: BannerProps) => {
     }
   }, [avatar]);
 
+  const [isFollowed, setIsFollowed] = useState(user?.isFollowed);
```
Comment (alexandrtovmach) [inline src/containers/Banner.tsx]: remove this state and use `currentUser.isFollowed`

File: src/containers/Banner.tsx [hunk 5]
```diff
@@ -151,6 +155,48 @@ const Banner = ({ isEditBanner, user }: BannerProps) => {
...
+  };
+
+  const [buttonText, setButtonText] = useState(initialText);
+
+  const userByUsernameId = useSelector<
+    RootState,
+    RootState["user"]["userByUsername"]["_id"]
+  >((store) => store.user.userByUsername._id);
+
+  const handleFollowClick = () => {
```
Comment (alexandrtovmach) [inline src/containers/Banner.tsx]: `event.preventDefault()` to avoid unnecessary page reload

File: src/redux/reducers/user.ts [hunk 6]
```diff
@@ -254,6 +275,30 @@ const userSlice = createSlice({
...
+    builder.addCase(addFollower.rejected, (store) => {
+      store.followedUser.isLoading = false;
+      store.followedUser.error = "Failed to fetch to follow user";
+    });
+
+    builder.addCase(deleteFollower.pending, (store) => {
+      store.followedUser.isLoading = true;
+    });
+    builder.addCase(deleteFollower.fulfilled, (store, { payload }) => {
+      store.followedUser = { ...payload };
```
Comment (alexandrtovmach) [inline src/redux/reducers/user.ts]: we don't need to keep this data in store

File: src/containers/Banner.tsx [hunk 7]
```diff
@@ -262,12 +308,15 @@ const Banner = ({ isEditBanner, user }: BannerProps) => {
...
+          <StyledButton
             className="rounded-pill fw-bold px-2 mt-3 me-3"
             variant="dark"
+            onClick={handleFollowClick}
+            onMouseOver={handleOnMouseOver}
+            onMouseLeave={handleOnMouseLeave}
           >
-            {t("common.follow")}
-          </Button>
+            {buttonText}
```
Comment (alexandrtovmach) [inline src/containers/Banner.tsx]: add spinner instead of text if `followedUser.isLoading` and disable button to prevent additional press

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

File: src/containers/Banner.tsx [hunk 8]
```diff
@@ -151,6 +155,48 @@ const Banner = ({ isEditBanner, user }: BannerProps) => {
     }
   }, [avatar]);
 
+  const [isFollowed, setIsFollowed] = useState(user?.isFollowed);
+
+  const initialText = () => {
+    if (isFollowed || false) {
```
Comment (VitaliiDenysiuk) [inline src/containers/Banner.tsx]:
```suggestion
    if (isFollowed) {
```
"|| false" nonsence

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**